src/recipe.cpp

```diff
@@ -729,32 +738,40 @@ std::set<proficiency_id> recipe::assist_proficiencies() const
     return ret;
 }
 
-float recipe::proficiency_time_maluses( const Character &guy ) const
+float recipe::proficiency_time_maluses( const Character &c ) const
 {
-    float total_malus = 1.0f;
+    double total_malus = 1.0;
     for( const recipe_proficiency &prof : proficiencies ) {
-        if( !guy.has_proficiency( prof.id ) &&
-            !helpers_have_proficiencies( guy, prof.id ) && prof.time_multiplier > 1.0f ) {
-            float malus = 1.0f + ( prof.time_multiplier - 1.0f ) *
-                          ( 1.0f - guy.crafting_inventory().get_book_proficiency_bonuses().time_factor( prof.id ) );
-            total_malus *= malus;
+        if( !c.has_proficiency( prof.id ) &&
+            !helpers_have_proficiencies( c, prof.id ) && prof.time_multiplier > 1.0f ) {
+            double malus = prof.time_multiplier - 1.0;
+            malus *= 1.0 - c.crafting_inventory().get_book_proficiency_bonuses().time_factor( prof.id );
+            double pl = get_aided_proficiency_level( c, prof.id );
+            // Sigmoid function that mitigates 100% of the time malus as pl approaches 1.0
+            // but has little effect at pl < 0.5. See #49198
+            malus *= 1.0 - std::pow( 0.5 + 0.5 * std::sin( pl * M_PI - M_PI / 2 ), 2 );
+            total_malus *= 1.0 + malus;
         }
     }
-    return total_malus;
+    return static_cast<float>( total_malus );
 }
 
-float recipe::proficiency_failure_maluses( const Character &guy ) const
+float recipe::proficiency_failure_maluses( const Character &c ) const
 {
-    float total_malus = 1.0f;
+    double total_malus = 1.0f;
     for( const recipe_proficiency &prof : proficiencies ) {
-        if( !guy.has_proficiency( prof.id ) &&
-            !helpers_have_proficiencies( guy, prof.id ) && prof.fail_multiplier > 1.0f ) {
-            float malus = 1.0f + ( prof.fail_multiplier - 1.0f ) *
-                          ( 1.0f - guy.crafting_inventory().get_book_proficiency_bonuses().fail_factor( prof.id ) );
-            total_malus *= malus;
+        if( !c.has_proficiency( prof.id ) &&
+            !helpers_have_proficiencies( c, prof.id ) && prof.fail_multiplier > 1.0f ) {
+            double malus =  prof.fail_multiplier - 1.0f;
+            malus *= 1.0 - c.crafting_inventory().get_book_proficiency_bonuses().fail_factor( prof.id );
+            double pl = get_aided_proficiency_level( c, prof.id );
+            // The failure malus is not completely eliminated until the proficiency is mastered.
+            // Most of the mitigation happens at higher pl. See #49198
+            malus *= 1.0 - ( 0.75 * std::pow( pl, 3 ) );
+            total_malus *= 1.0 + malus;
```

Comment:
Would it make sense to keep this formula contained to proficiency_set or a character function? e.g. would we want to use this formula somewhere else?